### PR TITLE
WT-12782 Increment the txn array entries walked count once instead of per iteration

### PR DESCRIPTION
**Pull request by y123456yz**
txn_walk_sessions optimize and __wt_rts_check bug fix

### CODE REVIEW

File: src/txn/txn.c [hunk 1]
```diff
@@ -256,9 +257,7 @@ __txn_get_snapshot_int(WT_SESSION_IMPL *session, bool update_shared_state)
 
     /* Walk the array of concurrent transactions. */
     WT_ACQUIRE_READ_WITH_BARRIER(session_cnt, conn->session_array.cnt);
-    WT_STAT_CONN_INCR(session, txn_walk_sessions);
```
Comment (luke-pearson) [inline src/txn/txn.c:259]: Keep this one here.

File: src/txn/txn.c [hunk 2]
```diff
@@ -299,6 +298,9 @@ __txn_get_snapshot_int(WT_SESSION_IMPL *session, bool update_shared_state)
         }
     }
```
Comment (luke-pearson) [inline src/txn/txn.c:300]: Remove this whitespace.

File: src/txn/txn.c [hunk 3]
```diff
@@ -418,9 +420,7 @@ __txn_oldest_scan(WT_SESSION_IMPL *session, uint64_t *oldest_idp, uint64_t *last
 
     /* Walk the array of concurrent transactions. */
     WT_ACQUIRE_READ_WITH_BARRIER(session_cnt, conn->session_array.cnt);
-    WT_STAT_CONN_INCR(session, txn_walk_sessions);
```
Comment (luke-pearson) [inline src/txn/txn.c:421]: Put this one back.

File: src/txn/txn.c [hunk 4]
```diff
@@ -465,6 +465,9 @@ __txn_oldest_scan(WT_SESSION_IMPL *session, uint64_t *oldest_idp, uint64_t *last
         }
     }
```
Comment (luke-pearson) [inline src/txn/txn.c:467]: Remove whitespace.

File: src/txn/txn.c [hunk 5]
```diff
@@ -2915,21 +2919,23 @@ __wt_verbose_dump_txn(WT_SESSION_IMPL *session)
      * handles is not thread safe, so some information may change while traversing if other threads
      * are active at the same time, which is OK since this is diagnostic code.
      */
-    WT_STAT_CONN_INCR(session, txn_walk_sessions);
     for (i = 0, s = txn_global->txn_shared_list; i < session_cnt; i++, s++) {
-        WT_STAT_CONN_INCR(session, txn_sessions_walked);
```
Comment (luke-pearson) [inline src/txn/txn.c:2920]: As we're in __wt_verbose_dump_txn perf isn't really an issue so i'd just leave the individual increments that way we don't need to introduce an  err: label and WT_ERR macros.

File: src/rollback_to_stable/rts.c [hunk 6]
```diff
@@ -44,11 +44,14 @@ __wt_rts_check(WT_SESSION_IMPL *session)
     WT_CONNECTION_IMPL *conn;
     WT_DECL_RET;
     WT_RTS_COOKIE cookie;
+    uint32_t session_cnt;
```
Comment (luke-pearson) [inline src/rollback_to_stable/rts.c:47]: Please revert the changes in this file, the `txn_sessions_walked` stat is meant to be used for traversals of the `txn_shared_list` this is traversing the conn->sessions array so it shouldn't increment these stats.

**Comment by y123456yz** [comment]
@luke-pearson 
I have change the code, please review again.
thanks.

File: src/txn/txn.c [hunk 7]
```diff
@@ -180,14 +181,14 @@ __wt_txn_active(WT_SESSION_IMPL *session, uint64_t txnid)
...
         /* If the transaction is in the list, it is uncommitted. */
         if (__wt_atomic_loadv64(&s->id) == txnid)
             goto done;
     }
 
     active = false;
 done:
+    WT_STAT_CONN_INCRV(session, txn_sessions_walked, i);
```
Comment (luke-pearson) [inline src/txn/txn.c:191]: Add a comment here:
/* We increment this stat here as the loop traversal can exit using a goto. */

**Comment by luke-pearson** [comment]
Could you also merge in develop?

**Comment by y123456yz** [comment]
@luke-pearson 
I have merge the develop and add the coment
thanks.

**Review by luke-pearson — APPROVED** [review]
LGTM.
